client/allocrunner/alloc_runner.go

```diff
@@ -473,6 +487,10 @@ func (ar *allocRunner) handleTaskStateUpdates() {
 			state := tr.TaskState()
 			states[name] = state
 
+			if tr.IsPoststopTask() {
+				continue
+			}
+
```

Comment:
Poststop tasks have special kill behavior: i.e. we don't want to kill them if we receive a kill signal from `nomad job stop`. We want to wait until everything else has been killed, then we want to run the poststop tasks.

Reply:
Skipping the rest of this loop essentially says: don't track poststop tasks along with the main group of tasks (in the set of  `liverunners`), that way we won't kill them later in a kill event scenario (i.e. `nomad job stop`).